examples/optuna/README.md

```diff
@@ -0,0 +1,87 @@
+# Example of Optuna-powered Hyper-parameters Tuning
+
+Tune hyper parameters using [Optuna](https://optuna.org/).
+
+Although the example script (`optuna_dqn_obs1d.py`) uses fixed target algorithm/environment
+(i.e., DQN and environments with 1d continuous observation space and discrete action space)
+in order to fucus on the concept of Optuna-powered PFRL,
+you can create one for your own use easily thanks to the Optuna's high flexibility!
+
+
+## How to Run
+
+### Quickstart
+
+The quickstart on your local machine.
+
+```bash
+storage="sqlite:///example.db"
+study="optuna-pfrl-quickstart"
+pruner="HyperbandPruner"
+
+# In common RL, higher score means better performance (`--direction maximize`)
+optuna create-study --study-name "${study}" --storage "${storage}" --direction maximize
+
+# Start tuning hyper parameters
+python optuna_dqn_obs1d.py --optuna-study-name "${study}" --optuna-storage "${storage}" --optuna-pruner "${pruner}"
+```
+
+You can see the optimization history of this study on [Jupyter Notebook](https://jupyter.org/install)
+via the [Optuna visualization module](https://optuna.readthedocs.io/en/latest/reference/visualization.html):
+
+```python
+# On jupyter notebook
+import optuna
+
+study_name = "optuna-pfrl-quickstart"
+storage = "sqlite:///example.db"
+study = optuna.load_study(study_name=study_name, storage=storage)
+
+optuna.visualization.plot_optimization_history(study)
+```
+
+![optimization_history](assets/optimization_history.png)
+
+
+If you are interested in the `--optuna-pruner` argument above, see the
+[corresponding Optuna document](https://optuna.readthedocs.io/en/latest/reference/pruners.html).
+
+
+### Distributed Optimization
+
+You might have already noticed that the sample script can be executed in parallel, and distributed:
+
+```bash
+# DB specs. We assume PostgreSQL here but you can use various backend DB engines.
+# Note that SQLite is not recommended for parallel optimization.
+postgres_user="user"
+postgres_password="password"
+postgres_host="host"
```

Comment:
I got `[E 2020-08-19 11:48:41,619] (psycopg2.OperationalError) could not translate host name "host" to address: nodename nor servname provided, or not known`. Does `host` need to be replaced with some value?

Reply:
These parameters are all dummies.
 I assumed that there exists `database` postgres db on some server accessible with name `host`, since setting up a postgres server requires some instructions that are not strongly related to optuna.

I'll add "If you can access a postgres server, ..."  or "For setting up a postgres server, see official instruction (some URL)".

Reply:
modified: https://github.com/pfnet/pfrl/pull/46/commits/3917f81b2b13a159be6d552b20a199f096bc57be